packages/hash/frontend/src/pages/entity-editor.page.tsx

```diff
@@ -16,7 +16,10 @@ import { useBlockProtocolFunctionsWithOntology } from "./type-editor/blockprotoc
  * Helper type-guard for determining if a `ValueOrArray` definition is an array or a value.
  */
 const isArrayDefinition = <T,>(input: ValueOrArray<T>): input is Array<T> =>
-  "type" in input && input.type === "array";
+  input &&
+  typeof input === "object" &&
+  "type" in input &&
+  input?.type === "array";
```

Comment:
@Alfred-Mountfield does this change make sense (saw you in `git blame`)? I was getting:

```
error TS2322: Type 'ValueOrArray<T>' is not assignable to type 'object'. Type 'T' is not assignable to type 'object'.
```

Reply:
Hm I wonder if we want to say `T extends object`, perhaps we don't

Reply:
Anyway I don't believe you need the `?` after `input?` as you already asserted it (can't suggest changes here for some reason)

Reply:
I removed `?` in https://github.com/hashintel/hash/pull/1534/commits/5209ef46dd69f52fd7c6cdce494af672cc6469f7, thanks for spotting! This should be flagged as an autofixable ESLint error when we get to enabling `@typescript-eslint/no-unnecessary-condition` ([internal Asana task](https://app.asana.com/0/1201095311341924/1202736501900834/f)).

Happy for you to switch to `T extends object` in a separate PR if you want. I assumed we could have ‘arrays’ of scalars (e.g. numbers or booleans) so did not change the base type.

Reply:
Enable alllll the lints :grin:

This isn't an array like `T[]`, it's an array _definition_ in JSON schema, so something like:
```
{ 
  "type": "array",
  "items":  {
    "$ref": "...",
  }
}